attestation-service/verifier/src/az_snp_vtpm/mod.rs

```diff
@@ -61,6 +61,19 @@ impl Verifier for AzSnpVtpm {
             bail!("unexpected empty report data");
         };
 
+        let mut expected_report_data = expected_report_data.to_vec();
+        match expected_report_data.len() {
```

Comment:
This `match` structure appear many times, maybe put it in a helper function like `regularize_report_data(expected_len: usize)` is better?

Reply:
nice idea.  Let me do this